package Nova;

message SuspectID_pb
{
	optional uint32 m_IP = 1;
	optional string m_ifname = 2;
}

enum ControlType
{
	CONTROL_EXIT_REQUEST = 0;			//Request for Novad to exit
	CONTROL_EXIT_REPLY = 1;				//Reply from Novad with success
	CONTROL_CLEAR_ALL_REQUEST = 2;		//Request for Novad to clear all suspects
	CONTROL_CLEAR_ALL_REPLY = 3;		//Reply from Novad with success
	CONTROL_CLEAR_SUSPECT_REQUEST = 4;	//Request for Novad to clear specified suspect
	CONTROL_CLEAR_SUSPECT_REPLY = 5;	//Reply from Novad with success
	CONTROL_SAVE_SUSPECTS_REQUEST = 6;	//Request for Novad to save the suspects list to persistent storage
	CONTROL_SAVE_SUSPECTS_REPLY = 7;	//Reply from Novad with success
	CONTROL_RECLASSIFY_ALL_REQUEST = 8;	//Request for Novad to reclassify all suspects
	CONTROL_RECLASSIFY_ALL_REPLY = 9;	//Reply from Novad with success
	CONTROL_CONNECT_REQUEST = 10;		//Request to connect to Novad from UI
	CONTROL_CONNECT_REPLY = 11;			//Reply from Novad with success
	CONTROL_DISCONNECT_NOTICE = 12;		//Notice to Novad that the UI is closing
	CONTROL_DISCONNECT_ACK = 13;
	CONTROL_START_CAPTURE = 14;			// Start packet capture on configured interfaces
	CONTROL_START_CAPTURE_ACK = 15;
	CONTROL_STOP_CAPTURE = 16;			// Stop packet capture on configured interfaces
	CONTROL_STOP_CAPTURE_ACK = 17;
	CONTROL_INVALID = 18;
};

//The different message types
enum RequestType
{
	// Requests for lists of suspect IPs
	REQUEST_SUSPECTLIST = 0;
	REQUEST_SUSPECTLIST_REPLY = 1;

	// Request for an individual suspect
	REQUEST_SUSPECT = 2;
	REQUEST_SUSPECT_REPLY = 3;

	REQUEST_SUSPECT_WITHDATA = 4;
	REQUEST_SUSPECT_WITHDATA_REPLY = 5;

	// Request for the uptime of novad
	REQUEST_UPTIME = 6;
	REQUEST_UPTIME_REPLY = 7;

	//Request to Novad to see if it's still alive
	REQUEST_PING = 8;
	REQUEST_PONG = 9;
};

enum UpdateType
{
	UPDATE_SUSPECT = 0;						//A new or updated suspect being sent to a UI
	UPDATE_SUSPECT_ACK = 1;					//Reply from Novad with success
	UPDATE_ALL_SUSPECTS_CLEARED = 2;		//A UI has cleared all suspect records
	UPDATE_ALL_SUSPECTS_CLEARED_ACK = 3;	//Acknowledgment of UPDATE_ALL_SUSPECTS_CLEARED
	UPDATE_SUSPECT_CLEARED = 4;				//A single suspect was cleared by a UI
	UPDATE_SUSPECT_CLEARED_ACK = 5;			//Acknowledgment of UPDATE_ALL_SUSPECTS_CLEARED_ACK
};

enum SuspectListType
{
	SUSPECTLIST_ALL = 0;
	SUSPECTLIST_HOSTILE = 1;
	SUSPECTLIST_BENIGN = 2;
};

message ControlMessage_pb
{
	required ControlType m_controlType = 1;
	optional string m_filePath = 2;
	optional bool m_success = 3;
	optional SuspectID_pb m_suspectID = 4;
}

message RequestMessage_pb
{
	required RequestType m_requestType = 1;
	optional SuspectListType m_listType = 2;
	repeated SuspectID_pb m_SuspectID = 3;
	optional uint32 m_startTime = 4;
}

message UpdateMessage_pb
{
	required UpdateType m_updateType = 1;
	optional SuspectID_pb m_SuspectID = 2;
}

